src/com/lftechnology/java/training/sanish/application/utility/UserInput.java

```diff
@@ -0,0 +1,172 @@
+     */
+    public static int getIntegerNumber(Scanner scanner, int min, int max) {
+        int intNum = 0;
+        boolean checkNextInput = true;
+        while (checkNextInput) {
+            if (!scanner.hasNextInt()) {
+                LOGGER.log(Level.WARNING, Constants.ENTER_VALID_NUMBER_MSG);
+                scanner.next();
+            } else {
+                intNum = scanner.nextInt();
+                if (min != -1 && max != -1) {
+                    if (intNum < min || intNum > max) {
+                        LOGGER.log(Level.WARNING, Constants.ENTER_NUMBER_RANGE_MSG + "({0}-{1}) : ", new Object[] { min, max });
+                    } else {
+                        checkNextInput = false;
+                    }
+                } else {
+                    checkNextInput = false;
+                }
+            }
+
+        }
+
+        return intNum;
+    }
+
+    /**
+     * Scan string form console and return string
+     *
+     * @param {@link Scanner}
+     * @return {@link String}
+     * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+     */
+    public static String getString(Scanner scanner) {
+        String inputString = "";
+        String pattern = "[a-zA-Z1-9!@#$%^&*/._,]+";
+        boolean checkNextInput = true;
+        while (checkNextInput) {
+            if (!scanner.hasNext()) {
+                scanner.next();
+            } else {
+                inputString = scanner.next();
+                if (inputString.isEmpty()) {
+                    continue;
+                } else if (!inputString.matches(pattern)) {
+                    LOGGER.log(Level.WARNING, Constants.INVALID_MSG);
+                } else {
+                    checkNextInput = false;
+                }
+            }
+        }
+
+        return inputString;
+    }
+
+    /**
+     * Scan email form console and return string
+     *
+     * @param {@link Scanner}
+     * @return {@link String}
+     * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+     */
+    public static String getEmail(Scanner scanner) {
+        String email = "";
+        String emailPattern = "[a-z][a-z_/+1-9]+@[a-z]+[.a-z]+";
+        while (true) {
+            if (!scanner.hasNext()) {
+                scanner.next();
+            } else {
+                email = scanner.next();
+                if (email.isEmpty()) {
+                    continue;
+                } else if (!email.matches(emailPattern)) {
+                    LOGGER.log(Level.WARNING, Constants.INVALID_EMAIL_MSG);
+                } else {
+                    break;
+                }
+            }
+        }
+
+        return email;
+    }
+
+    /**
+     * Scan user role number
+     *
+     * @param {@link Scanner}
+     * @return {@link String}
+     * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+     */
+    public static String getUserRole(Scanner scanner) {
+        String role;
+        int intNum = 1;
+        while (true) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS135)